### PR TITLE
feat(firestore-counter): convert Firestore Counter to use scheduled functions

### PR DESCRIPTION
**Pull request by Ehesp**
Converts the Firestore Counter extension to scheduled functions & removes legacy HTTP endpoints.

### CODE REVIEW

**Comment by Ehesp** [comment]
@laurenzlong What's the upgrade flow for users with these changes? Thinking out loud, we have the following things which would be effected:

- The `controller` legacy HTTP request endpoint currently forwards a PubSub message on to trigger the `controllerCore` handler (which is now a scheduled function). Are we safe to remove this (existing HTTP requests will start to fail).
- If we're changing `controllerCore` to be a scheduled function, existing Cloud Scheduler jobs will still exist but start to fail. 

With these changes, we'd be safe to remove these docs:

image

However, should we have some sort of "Upgrading from 0.1.3 -> 0.1.4" section which instead suggests that the user should remove any existing Cloud Scheduler jobs / HTTP trigger requests?

File: firestore-counter/extension.yaml [hunk 1]
```diff
@@ -140,13 +132,8 @@ params:
...
-      How often should the function to aggregatte shards be run?
-    options:
-      - label: every 1 minutes
-        value: every 1 minutes
-      - label: every 2 minutes
-        value: every 2 minutes
-      - label: every 5 minutes
-        value: every 5 minutes
-    default: every 1 minutes
+      In minutes, how often should the function to aggregatte shards be run?
```
Comment (russellwheatley) [inline firestore-counter/extension.yaml:135]: spelling -> `aggregate`

**Comment by Ehesp** [comment]
@laurenzlong :

Can we remove the HTTP endpoint (and pubsub trigger) as part of this PR?

**Comment by jhuleatt** [comment]
> Can we remove the HTTP endpoint (and pubsub trigger) as part of this PR?

Yes. We can add a warning to the update flow that explains the change.

**Comment by russellwheatley** [comment]
@googlebot I fixed it.

**Comment by russellwheatley** [comment]
@googlebot I consent.

**Comment by russellwheatley** [comment]
@jhuleatt could you check the `go here` info link and let me know what I need to do to pass cla, please? I don't have a google account so I'm blocked from accessing the link. Thanks!

**Comment by joehan** [comment]
@russellwheatley I just checked the `go here` link, and it just has a list of who has and has not signed the CLA. I think you need to go to https://cla.developers.google.com/clas and accept the Google Corporate SLA, and then comment `@googlebot I fixed it` here to have google bot recheck for you.

**Comment by dackers86** [comment]
Re: Installations errors during testing. 

```js
Error: ; RESOURCE_ERROR at /deployments/firebase-ext-firestore-counter/resources/firebase-ext-firestore-counter-controllerCore: {“ResourceType”:“gcp-types/cloudscheduler-v1:projects.locations.jobs”,“ResourceErrorCode”:“403",“ResourceErrorMessage”:{“code”:403,“message”:“The principal (user or service account) lacks IAM permission \“cloudscheduler.jobs.create\” for the resource \“projects/extensions-testing/locations/europe-west\” (or the resource may not exist).“,”status”:“PERMISSION_DENIED”,“statusMessage”:“Forbidden”,“requestPath”:“https://cloudscheduler.googleapis.com/v1/projects/extensions-testing/locations/europe-west/jobs”,“httpMethod”:“POST”}}
```

@tinaliang @joehan have been added to a working development (Firebase) example.

Let me know if you need anything else with this

**Comment by tinaliang** [comment]
> Re: Installations errors during testing.
> 
> ```js
> Error: ; RESOURCE_ERROR at /deployments/firebase-ext-firestore-counter/resources/firebase-ext-firestore-counter-controllerCore: {“ResourceType”:“gcp-types/cloudscheduler-v1:projects.locations.jobs”,“ResourceErrorCode”:“403",“ResourceErrorMessage”:{“code”:403,“message”:“The principal (user or service account) lacks IAM permission \“cloudscheduler.jobs.create\” for the resource \“projects/extensions-testing/locations/europe-west\” (or the resource may not exist).“,”status”:“PERMISSION_DENIED”,“statusMessage”:“Forbidden”,“requestPath”:“https://cloudscheduler.googleapis.com/v1/projects/extensions-testing/locations/europe-west/jobs”,“httpMethod”:“POST”}}
> ```
> 
> @tinaliang @joehan have been added to a working development (Firebase) example. You should have received emails on 23/09.
> 
> Let me know if you need anything else.

@dackers86 I'm getting `{"error":{"code":403,"message":"Policy update access denied.","status":"PERMISSION_DENIED"}}` when trying to install an extension on this project. What permission were we granted to the `extensions-testing` project?

**Comment by dackers86** [comment]
Hi @tinaliang.

You originally had editor access, I have bumped this up to owner.

An invite has been sent to your email addresses

**Comment by tinaliang** [comment]
@dackers86 I was just able to successfully install an instance of this extension in the `extensions-testing` project. Can you confirm?

**Comment by dackers86** [comment]
Hi @tinaliang.

We do see a new extension installed...

image

You're unique number for the installation is `ftob`

**Comment by tinaliang** [comment]
@dackers86 It seems like the installation errors are fixed then?

**Comment by Ehesp** [comment]
I'll double check my side (since I originally had the error).

**Comment by Ehesp** [comment]
Ok looks like it's installing with no errors now, not sure what was wrong!

**Review by russellwheatley — APPROVED** [review]
LGTM

**Review by jhuleatt — APPROVED** [review]
@tinaliang as part of our release review this week, can you please test the update flow from an older version of this extension (to make sure that the change from pubsub trigger to scheduled function doesn't break anything)?

**Review by tinaliang — APPROVED** [review]

**Comment by tinaliang** [comment]
I was able to successfully update the extension!

**Pull request merged**